src/js/components/settings/billing.js

```diff
@@ -0,0 +1,227 @@
+import React from 'react';
+import moment from 'moment';
+
+// material ui
+import { Button, Divider, LinearProgress, Tooltip } from '@material-ui/core';
+import WarningIcon from '@material-ui/icons/Warning';
+
+import AppStore from '../../stores/app-store';
+import AppActions from '../../actions/app-actions';
+import MonthlyBillingInformation from './monthlybillinginformation';
+import PlanNotification from './plannotification';
+
+const totalFreeCredit = 120;
+
+const interactionList = [
+  {
+    title: 'Devices',
+    billingInformation: [
+      {
+        title: 'Device activations:',
+        key: 'activation',
+        explanation: '# new devices connected to the server'
+      },
+      {
+        title: 'Total active devices:',
+        key: 'monthly_device',
+        explanation: '# devices that were active during this period'
+      }
+    ]
+  },
+  {
+    title: 'Deployments',
+    billingInformation: [
+      {
+        title: 'Device deployments:',
+        key: 'device_deployment',
+        explanation: '# devices targeted by a deployment'
+      }
+    ]
+  },
+  {
+    title: 'Data storage',
+    billingInformation: [
+      {
+        title: 'GB of data stored:',
+        key: 'data_storage',
+        explanation: 'The software you have uploaded to the hosted Mender service'
+      }
+    ]
+  },
+  {
+    title: 'Data transfer',
+    billingInformation: [
+      {
+        title: 'GB of data transferred:',
+        key: 'data_transfer',
+        explanation: 'Data traffic that was sent between a device the hosted Mender service'
+      }
+    ]
+  }
+];
+
+class Billing extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      showUsage: false,
+      billingInformation: {
+        interactions: [],
+        timeframe: { month: 0, year: 0 },
+        timestamp: new Date(),
+        total: 0
+      }
+    };
+  }
+
+  componentDidMount() {
+    this.updateBillingStatement();
+  }
+
+  updateBillingStatement(date = new Date()) {
+    const self = this;
+    const timeframe = { month: date.getMonth(), year: date.getFullYear() };
+    AppActions.getBillingStatus(timeframe).then(information => {
+      const interactions = interactionList.map(item => {
+        item.billingInformation = item.billingInformation.map(infoItem => ({
+          ...infoItem,
+          unitFee: information[`${infoItem.key}_unit_fee`],
```

Comment:
The values should probably be rounded otherwise long values are breaking the layout a bit. Maybe to 4 decimal places? It may occasionally result a penny difference here or there, but I think for usability and readability it makes sense

Reply:
right, I assumed those came ready from the backend - should be limited to 4 decimal places if they are available